rectools/metrics/base.py

```diff
@@ -92,3 +92,43 @@ def merge_reco(reco: pd.DataFrame, interactions: pd.DataFrame) -> pd.DataFrame:
         how="left",
     )
     return merged
+
+
+def outer_merge_reco(reco: pd.DataFrame, interactions: pd.DataFrame) -> pd.DataFrame:
+    """
+    Merge recommendation table with interactions table with outer join. All ranks for all users are
+    present with no skipping. Null ranks will be specified for test interactions that were not
+    predicted in recommendations.
+    This method is useful for AUC based ranking metrics.
+
+    Parameters
+    ----------
+    reco : pd.DataFrame
+        Recommendations table with columns `Columns.User`, `Columns.Item`, `Columns.Rank`.
+    interactions : pd.DataFrame
+        Interactions table with columns `Columns.User`, `Columns.Item`.
+
+    Returns
+    -------
+    pd.DataFrame
+        Result of merging with added `__test_positive` boolean column.
+    """
+    prepared_interactions = interactions.drop_duplicates(Columns.UserItem).reindex(columns=Columns.UserItem).copy()
```

Comment:
Fixed.
Without a copy we will show `A value is trying to be set on a copy of a slice from a DataFrame.`
Because we dropped some rows and then assigned ["__test_positive"] = True

Reply:
I was sure `reindex` creates copy itself

Reply:
Checked again. You are right, I'll drop the `copy`

Reply:
fixed